### ISSUE TITLE
Heroku error message on free account ECONNREFUSED

### ISSUE THREAD

**Issue by alexxgathp** [strongloop/loopback-connector-mysql#108]
I am trying out the Heroku free account with strongloop for fun. My local server works, but Heroku is a no go.

I'm getting this error from the loopback browser explorer on the Heroku free account I setup, ran successfully "heroku open" after following the other steps sequentially:

**Error message in loopback explorer browser**
```javascript
{
  "error": {
    "name": "Error",
    "status": 500,
    "message": "connect ECONNREFUSED",
    "code": "ECONNREFUSED",
    "errno": "ECONNREFUSED",
    "syscall": "connect",
    "fatal": true,
    "stack": "Error: connect ECONNREFUSED\n    at exports._errnoException (util.js:746:11)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1010:19)\n    --------------------\n    at Protocol._enqueue (/app/node_modules/loopback-connector-mysql/node_modules/mysql/lib/protocol/Protocol.js:135:48)\n    at Protocol.handshake (/app/node_modules/loopback-connector-mysql/node_modules/mysql/lib/protocol/Protocol.js:52:41)\n    at PoolConnection.connect (/app/node_modules/loopback-connector-mysql/node_modules/mysql/lib/Connection.js:123:18)\n    at Pool.getConnection (/app/node_modules/loopback-connector-mysql/node_modules/mysql/lib/Pool.js:45:23)\n    at MySQL.executeSQL (/app/node_modules/loopback-connector-mysql/lib/mysql.js:199:12)\n    at /app/node_modules/loopback-connector-mysql/node_modules/loopback-connector/lib/sql.js:408:10\n    at /app/node_modules/loopback-datasource-juggler/lib/observer.js:175:9\n    at doNotify (/app/node_modules/loopback-datasource-juggler/lib/observer.js:93:49)\n    at MySQL.ObserverMixin._notifyBaseObservers (/app/node_modules/loopback-datasource-juggler/lib/observer.js:116:5)\n    at MySQL.ObserverMixin.notifyObserversOf (/app/node_modules/loopback-datasource-juggler/lib/observer.js:91:8)"
  }
}
```

**Package.json**
```javascript
{
  "name": "vapserver",
  "version": "1.0.0",
  "main": "server/server.js",
  "scripts": {
    "pretest": "jshint ."
  },
  "engines": {
        "node": "0.12.x"
    },
  "dependencies": {
    "compression": "^1.0.3",
    "cors": "^2.5.2",
    "errorhandler": "^1.1.1",
    "loopback": "^2.14.0",
    "loopback-boot": "^2.6.5",
     "loopback-datasource-juggler": "latest",
    "loopback-connector-mysql": "latest",
    "strong-express-metrics": "^1.1.0",
    "serve-favicon": "^2.0.1",
	"ecdsa": "latest"
  },
  "optionalDependencies": {
    "loopback-explorer": "^1.1.0"
  },
  "devDependencies": {
    "jshint": "^2.5.6"
  },
  "repository": {
    "type": "",
    "url": ""
  },
  "description": "vapserver"
}
```

**config.json**
```javascript
{
  "restApiRoot": "/api",
  "remoting": {
    "context": {
      "enableHttpContext": false
    },
    "rest": {
      "normalizeHttpPath": false,
      "xml": false
    },
    "json": {
      "strict": false,
      "limit": "50mb"
    },
    "urlencoded": {
      "extended": true,
      "limit": "50mb"
    },
    "cors": false,
    "errorHandler": {
      "disableStackTrace": false
    }
  },
  "legacyExplorer": false
}
```

**datasources.json**
```javascript
{
  "db": {
    "name": "db",
    "connector": "memory"
  },
  "media_history": {
  "host": "remote hostname",
  "database": "database name",
    "username": "username",
    "password": "password",
    "name": "media_history",
    "connector": "mysql"
  }
}
```

**Comment by alexxgathp** [comment]
Hmmmm, I got same error message when I deployed it on Amazone EC2. This may be the database server having an issue of some sort, will explore a little more...

**Comment by alexxgathp** [comment]
Okay, I figured it out. It was super-counter-intuitive at first.

The datasources.json was the culprit. I had stopped tracking it in my Git, since I didn't want the password and username info tracked on Git, so the old non-configured datasources.json was getting sent to Amazon EC2, and I imagine Heroku too.

I can see how to change it on Amazon easily via SSH, but is there a way to push the newer one without tracking it to the Heroku?

So what I want is something like a "git push heroku master" that sends the newer datasources.json without actually tracking it in the git, since I don't want it to go into the repository. I'll close this issue, but feel free to comment on Heroku advice. 

I'd rather use their free account than Amazon's free account, since Amazon requires my credit card for a free account...
